test/domain_and_path_test.js

```diff
@@ -186,8 +186,8 @@ vows
       },
       "trailing slash": {
         topic: tough.permutePath.bind(null, '/foo/bar/'),
-        "three things": function (list) {
-          assert.deepEqual(list, ['/foo/bar', '/foo', '/']);
+        "four things": function (list) {
+          assert.deepEqual(list, ['/foo/bar/', '/foo/bar', '/foo', '/']);
```

Comment:
Perhaps `/foo/` should be an intermediate as well? 

Overall, I think I have a performance concern: is this is going to result in a lot more hits to the underlying store?